File: Plugins/Aspose_Tasks_Java_for_Ruby/lib/asposetasksjava/TaskLinks/definelinktype.rb
module Asposetasksjava
  module DefineLinkType
    def initialize()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'
                
        # Instantiate project object
        project = Rjb::import('com.aspose.tasks.Project').new

    	pred = project.getRootTask().getChildren().add("Task 1")
    	succ = project.getRootTask().getChildren().add("Task 2")
    	link = project.getTaskLinks().add(pred, succ)
        link.setLinkType(Rjb::import('com.aspose.tasks.TaskLinkType').StartToStart)

		puts "Defined task link type."
    end
  end
end

File: Plugins/Aspose_Tasks_Java_for_PHP/src/aspose/tasks/WorkingWithTaskLinks/DefineLinkType.php
<?php
namespace Aspose\Tasks\WorkingWithTaskLinks;

use com\aspose\tasks\Project as Project;
use com\aspose\tasks\TaskLinkType as TaskLinkType;
class DefineLinkType{

    public static function run($dataDir=null){

        # Instantiate project object
        $project = new Project();

        $pred = $project->getRootTask()->getChildren()->add("Task 1");
    	$succ = $project->getRootTask()->getChildren()->add("Task 2");
    	$link = $project->getTaskLinks()->add($pred, $succ);
        $taskLinkType=new TaskLinkType();
        $link->setLinkType($taskLinkType->StartToStart);

		print "Defined task link type.".PHP_EOL;
    }

}
?>